homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,224 @@
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['TodayMin']):
+                        self.session.data.minmax[nodeid]['TodayMin'] = \
+                            curtempret
+
+                    if (curtempret > self.session.data.minmax[nodeid]
+                            ['TodayMax']):
+                        self.session.data.minmax[nodeid]['TodayMax'] = \
+                            curtempret
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['RestartMin']):
+                        self.session.data.minmax[nodeid]['RestartMin'] = \
+                            curtempret
+
+                    if (curtempret >
+                            self.session.data.minmax[nodeid]
+                            ['RestartMax']):
+                        self.session.data.minmax[nodeid]['RestartMax'] = \
+                            curtempret
+                else:
+                    current_node_max_min_data = {}
+                    current_node_max_min_data['TodayMin'] = curtempret
+                    current_node_max_min_data['TodayMax'] = curtempret
+                    current_node_max_min_data['TodayDate'] = \
+                        datetime.date(datetime.now())
+                    current_node_max_min_data['RestartMin'] = curtempret
+                    current_node_max_min_data['RestartMax'] = curtempret
+                    self.session.data.minmax[nodeid] = \
+                        current_node_max_min_data
+            else:
+                curtempret = 0
+            return curtempret
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        set_result = None
+
+        if self.device_type == "Heating":
+            set_result = self.session.heating.get_target_temperature(
+                self.node_id)
+        elif self.device_type == "HotWater":
+            set_result = None
+
+        return set_result
+
+    @property
+    def min_temp(self):
+        """Return minimum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.min_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.max_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def operation_list(self):
+        """List of the operation modes."""
+        return self.modes
+
+    @property
+    def current_operation(self):
+        """Return current mode."""
+        if self.device_type == "Heating":
+            currentmode = self.session.heating.get_mode(self.node_id)
+            if currentmode == "SCHEDULE":

```

Comment:
Use a dictionary that maps the hive state to the home assistant state. Then you don't need to have these long if/elif branches.

Reply:
I don't understand how to do that

Reply:
```py
if self.device_type == "Heating":
    currentmode = self.session.heating.get_mode(self.node_id)
elif self.device_type == "HotWater":
    currentmode = self.session.hotwater.get_mode(self.node_id)
else:
    return None
HIVE_TO_HASS_STATE = {
    'SCHEDULE': STATE_AUTO, 'MANUAL': STATE_HEAT, ...}
return HIVE_TO_HASS_STATE.get(current_mode)
```

Reply:
thanks, but I am not sure that will work as the heating entities have `auto, heat, off` and the hotwater entities have `auto, on, off`

Reply:
I don't see any conflicts between the would be keys in the dictionary, ie, `'SCHEDULE'`, `'MANUAL'` etc.

Reply:
ok, I will take another look.

Reply:
thanks. done